test/integration/src/main/java/io/pravega/test/integration/utils/TestUtils.java

```diff
@@ -0,0 +1,212 @@
+        StreamManager streamManager = StreamManager.create(clientConfig);
+        assertNotNull(streamManager);
+
+        boolean result = streamManager.createScope(scope);
+        for (String stream: streams) {
+            boolean isStreamCreated = streamManager.createStream(scope, stream,
+                    StreamConfiguration.builder().scalingPolicy(ScalingPolicy.fixed(1)).build());
+            result = result && !isStreamCreated ? false : true;
+        }
+        return result;
+    }
+
+    /**
+     * Write the {@code message} to the specified {@code scope}/{@code stream}, using the
+     * provided {@writerClientConfig}.
+     *
+     * @param scope the scope
+     * @param stream the stream
+     * @param message the message to write. If it is null, a default message will be used.
+     * @param writerClientConfig the {@link ClientConfig} object to use to connect to the server
+     * @throws NullPointerException if {@code scope} or {@code stream} or {@writerClientConfig} is null
+     * @throws RuntimeException if any exception is thrown by the client
+     */
+    @SneakyThrows
+    public static void writeDataToStream(String scope, String stream, String message, ClientConfig writerClientConfig) {
+        writeDataToStream(scope, stream, message, 1, writerClientConfig);
+    }
+
+    /**
+     * Write the specified number of messages to the specified {@code scope}/{@code stream}, using the
+     * provided {@writerClientConfig}.
+     *
+     * @param scope the scope
+     * @param stream the stream
+     * @param message the event message to write. If it is null, a default message will be used.
+     * @param numMessages the number of event messages to write
+     * @param writerClientConfig the {@link ClientConfig} object to use to connect to the server
+     * @throws NullPointerException if {@code scope} or {@code stream} or {@writerClientConfig} is null
+     * @throws IllegalArgumentException if {@code numMessages} < 1
+     * @throws RuntimeException if any exception is thrown by the client
+     */
+    public static void writeDataToStream(@NonNull String scope, @NonNull String stream, String message, int numMessages,
+                                         @NonNull ClientConfig writerClientConfig) {
+        Preconditions.checkArgument(numMessages > 0);
+        if (message == null) {
+            message = "Test message";
+        }
+        @Cleanup final EventStreamClientFactory writerClientFactory = EventStreamClientFactory.withScope(scope,
+                writerClientConfig);
+
+        @Cleanup final EventStreamWriter<String> writer = writerClientFactory.createEventWriter(stream,
+                new JavaSerializer<String>(),
+                EventWriterConfig.builder().build());
+        List<CompletableFuture<Void>> futures = new ArrayList<>();
+        for (int i = 0; i < numMessages; i++) {
+            futures.add(writer.writeEvent(message));
+        }
+        Futures.allOf(futures).join();
```

Comment:
we can simply use a writer.flush() here.

Reply:
Thanks. That change makes the code much simpler. I've made the change as suggested.